src/openrct2/world/MapGen.cpp

```diff
@@ -134,46 +138,43 @@ void mapgen_generate_blank(mapgen_settings* settings)
 
 void mapgen_generate(mapgen_settings* settings)
 {
-    int32_t x, y, mapSize, floorTexture, wallTexture, waterLevel;
-
-    mapSize = settings->mapSize;
-    floorTexture = settings->floor;
-    wallTexture = settings->wall;
-    waterLevel = settings->water_level;
-
-    if (floorTexture == -1)
+    auto mapSize = settings->mapSize;
+    auto waterLevel = settings->water_level;
+    const auto selectedFloor = TerrainSurfaceObject::GetById(settings->floor);
+    std::string floorTexture = selectedFloor != nullptr ? std::string(selectedFloor->GetIdentifier()) : "";
+    const auto selectedEdge = TerrainEdgeObject::GetById(settings->wall);
+    std::string edgeTexture = selectedFloor != nullptr ? std::string(selectedEdge->GetIdentifier()) : "";
+
+    if (floorTexture == "")
```

Comment:
```suggestion
    if (floorTexture.empty())
```

Reply:
... why? It's probably a matter of taste, but I find `== ""` to be much more recognisable.

Reply:
Although i doubt it makes any difference .empty() will likely be faster to compute.